cdap-watchdog/src/main/java/co/cask/cdap/logging/context/LoggingContextHelper.java

```diff
@@ -115,6 +117,20 @@ public static LoggingContext getLoggingContext(Map<String, String> tags) {
                                        tags.get(ApplicationLoggingContext.TAG_RUN_ID),
                                        tags.get(ApplicationLoggingContext.TAG_INSTANCE_ID));
     } else if (tags.containsKey(WorkflowLoggingContext.TAG_WORKFLOW_ID)) {
+      if (tags.containsKey(WorkflowProgramLoggingContext.TAG_WORKFLOW_MAP_REDUCE_ID)) {
+        return new WorkflowProgramLoggingContext(namespaceId, applicationId,
+                                                 tags.get(WorkflowLoggingContext.TAG_WORKFLOW_ID),
+                                                 tags.get(ApplicationLoggingContext.TAG_RUN_ID), ProgramType.MAPREDUCE,
```

Comment:
Yes, actually when the WorkflowProgramLoggingContext is being created in BasicMapReduceContext and MapReduceClassLoader, the run id is being set to the workflow's run id. @sagarkapare Why is it not set to the MR's run id?